src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -528,6 +545,36 @@ private static string ComposeMessageFromInnerExceptions(Exception exception)
             }
         }
 
+        private RuntimeGraph LoadRuntimeGraph(LockFile lockFile)
```

Comment:
Taking a look at the apis available in NuGet this seems right. 

@emgarten would you mind taking a quick look at this?

@anurse , @pakrym I am looking through the WorkspaceContext but not too sure how to integrate this, could you provide some guidance there?